src/main/java/in/org/projecteka/hiu/dataprocessor/DiagnosticReportResourceProcessor.java

```diff
@@ -48,10 +57,15 @@ private void processMedia(DiagnosticReport diagnosticReport, Path localStoragePa
             return;
         }
 
+        boolean radiologyCategory = isRadiologyCategory(diagnosticReport);
+
         for (DiagnosticReport.DiagnosticReportMediaComponent media : mediaList) {
             if (media.hasLink()) {
                 Media linkTarget = (Media) media.getLink().getResource();
-                saveAttachmentAsFile(linkTarget.getContent(), localStoragePath);
+                Path savedAttachmentPath = saveAttachmentAsFile(linkTarget.getContent(), localStoragePath);
+                if (radiologyCategory && isRadiologyFile(linkTarget.getContent())) {
+                    uploadToLocalDicomServer(linkTarget.getContent(), savedAttachmentPath);
+                }
```

Comment:
I think we have to handle `saveAttachmentAsFile` returns null scenario too. 

Instead of returning Path, you could return Optional<Path> I believe.

Reply:
Apoorva is working on the same code, and added some error handling. We will do that together as part of her PR.